frontend/packages/otr/src/redux/sagas/clerkInterpreterOverview.ts

```diff
@@ -82,10 +86,41 @@ function* updateClerkInterpreterOverview(
   }
 }
 
+function* doUpdateQualificationPublishPermission(
+  action: PayloadAction<Qualification>
+) {
+  try {
+    const apiResponse: AxiosResponse<ClerkInterpreterResponse> = yield call(
+      axiosInstance.put,
+      APIEndpoints.Qualification,
+      SerializationUtils.serializeQualification({
+        ...action.payload,
+        permissionToPublish: !action.payload.permissionToPublish,
+      })
+    );
+    const interpreter = SerializationUtils.deserializeClerkInterpreter(
+      apiResponse.data
+    );
+    yield put(storeClerkInterpreterOverview(interpreter));
+    yield put(updateQualificationPublishPermissionSuccess());
+  } catch (error) {
+    yield put(rejectQualificationPublishPermissionUpdate());
+    yield put(
+      showNotifierToast(
+        NotifierUtils.createAxiosErrorNotifierToast(error as AxiosError)
+      )
+    );
+  }
+}
+
 export function* watchFetchClerkInterpreterOverview() {
   yield takeLatest(loadClerkInterpreterOverview, fetchClerkInterpreterOverview);
   yield takeLatest(
     updateClerkInterpreterDetails,
     updateClerkInterpreterOverview
   );
+  yield takeLatest(
+    updateQualificationPublishPermission,
+    doUpdateQualificationPublishPermission
+  );
```

Comment:
Jos tämän siirtäisi qualification sagan alle, niin saiskohan sinne samalla lailla yleisemmän watch funktion `watchQualificationUpdates()` tms. jonka alla olisi määritelty tämä yield sekä `addQualification` että `removeQualification` logiikat

Reply:
Sinne jos tämän siirtää niin nimeksi `onUpdateQualificationPublishPermission` tms.

Reply:
Itse asiassa, oiskohan parempi kuitenkin nimetä tämä metodi sekä vast. status reducerin alla yms. olemaan `updateQualification` (ja ylle `onQualificationUpdate`). Sen sijaan että ylempänä määriteltäisiin, että serialisoitavassa datassa vaihdetaan `permisionToPublish` flagin arvo, tehtäisiin tuo jo tätä update-metodia kutsuvassa funktiossa. Näin tämä toimisi yleiskäyttöisenä rekisteröinnin päivitysmetodina kun sitä bäkkärin päästäkin käytetään.

Reply:
Jep, siirretty yhteisen alle seuraavasti

`
export function* watchQualificationUpdates() {
  yield takeLatest(addQualification, onAddQualification);
  yield takeLatest(removeQualification, onRemoveQualification);
  yield takeLatest(updateQualification, onUpdateQualification);
}
`